citrination_client/base/base_client.py

```diff
@@ -57,6 +58,18 @@ def _handle_response(self, response, failure_message=DEFAULT_FAILURE_MESSAGE):
         check_general_success(response, failure_message)
         return response
 
+    def _wait_for(self, activity_description, criterion, timeout, check_delay=5):
+        start = time.time()
+        while time.time() - start < timeout:
+            time.sleep(check_delay)
+            try:
+                if criterion():
+                    return
+            except Exception as e:
+                raise RuntimeWarning("Error checking wait criterion: {}".format(e))
```

Comment:
Are you sure you meant to raise RuntimeWarning here? I'm fairly sure this will exit on exceptions (note - these exceptions can be blips/timeouts/etc.).

Suggestion: `warnings.warn("Error checking wait criterion: {}".format(e)", RuntimeWarning))`

Reply:
This has changed from the last time I saw it...
The basic idea was to copy the `_wait_for()` from citrination-workflow-tests:
```
start = time.time()
        while time.time() - start < timeout:
            time.sleep(check_delay)
            try:
                if criterion():
                    return
            except Exception as e:
                print("Error checking wait criterion: {}".format(e))

        raise RuntimeError("Failed to `{}` in {} seconds".format(activity_description, timeout))
```
But this would throw an Error even if it was just a timeout, and the process could still succeed. Basically I didn't want the functions using `_wait_for()` to throw an error when it's just a timeout. I don't want a timeout to block the `return` of an `upload_result`, for example. So yes, `warnings.warn` would be good. But this is now making me reconsider how these are implemented in general.

Reply:
Yeah I probably should have asked for more clarification before changing this. There was a request to make this throw an error/exception, so I changed it to that, and another (unrelated) request to not send more things to stdout (which the current built-in warning function does). 

Agreed that `warnings.warn` makes sense.